src/ir/possible-contents.cpp

```diff
@@ -896,6 +896,27 @@ struct InfoCollector
     }
     addRoot(curr, PossibleContents::exactType(curr->type));
   }
+  void visitArrayNewSeg(ArrayNewSeg* curr) {
+    if (curr->type == Type::unreachable) {
+      return;
+    }
+    auto heapType = curr->type.getHeapType();
+    switch (curr->op) {
+      case NewData: {
+        Type elemType = heapType.getArray().element.type;
+        addRoot(DataLocation{heapType, 0},
+                PossibleContents::exactType(elemType));
```

Comment:
```suggestion
                PossibleContents::fromType(elemType));
```

Reply:
(we don't know this is an exact type, do we? regardless the `fromType` helper is meant for this: a situation where all you know is the wasm type)

Reply:
(Ah, it's an exact type since it's coming from a data section... makes sense. Still, please use the helper.)